testsuite/MDAnalysisTests/lib/test_distances.py

```diff
@@ -43,3 +43,92 @@ def test_transform_StoR_fail():
 
     with pytest.raises(TypeError, match='S must be of type float32'):
         r = mda.lib.distances.transform_StoR(s, box)
+
+
+# different boxlengths to shift a coordinate
+shifts = [
+    ((0, 0, 0), (0, 0, 0), (0, 0, 0), (0, 0, 0)), # no shifting
+    ((1, 0, 0), (0, 1, 1), (0, 0, 1), (1, 1, 0)), # single box lengths
+    ((-1, 0, 1), (0, -1, 0), (1, 0, 1), (-1, -1, -1)), # negative single
+    ((4, 3, -2), (-2, 2, 2), (-5, 2, 2), (0, 2, 2)),  # multiple boxlengths
+]
+
+
+@pytest.mark.parametrize('shift', shifts)
+@pytest.mark.parametrize('periodic', [True, False])
+def test_calc_distance(shift, periodic):
+    box = np.array([10, 10, 10, 90., 90., 90.], dtype=np.float32)
+
+    coords = np.array([[1, 1, 1], [3, 1, 1]], dtype=np.float32)
+
+    shift1, shift2, _, _ = shift
+
+    coords[0] += shift1 * box[:3]
+    coords[1] += shift2 * box[:3]
+
+    box = box if periodic else None
+    result = mda.lib.distances.calc_distance(coords[0], coords[1], box)
+
+    reference = 2.0 if periodic else np.linalg.norm(coords[0] - coords[1])
+
+    assert result == pytest.approx(reference)
+
+
+@pytest.mark.parametrize('case', [
+    # 90 degree angle
+    (np.array([[1, 1, 1], [1, 2, 1], [2, 2, 1]], dtype=np.float32), 90.),
+    # straight line / 180.
+    (np.array([[1, 1, 1], [1, 2, 1], [1, 3, 1]], dtype=np.float32), 180.),
```

Comment:
90 and 180° are special cases. Maybe something an angle less peculiar would be worth testing.